### PR TITLE
Add rails 7.1 support

### PR DESCRIPTION
**Pull request by jamondrag**
Add rails 7.1 support in gemspec.

Specs pass locally.

Screenshot 2024-01-16 at 6 20 50 PM

### CODE REVIEW

File: activerecord-delay_touching.gemspec [hunk 1]
```diff
@@ -18,5 +18,5 @@ Gem::Specification.new do |spec|
   spec.test_files    = spec.files.grep(%r{^(test|spec|features)/})
   spec.require_paths = ["lib"]
 
-  spec.add_dependency             "activerecord", ">= 4.2", "< 7.1"
+  spec.add_dependency             "activerecord", ">= 4.2", "~> 7.1"
```
Comment (nefuhs) [inline activerecord-delay_touching.gemspec]: Why this change in particular, as opposed to `"activerecord", ">= 4.2", "< 7.2"`?

Comment (jamondrag) [reply activerecord-delay_touching.gemspec]: Sorry just seeing this.. No reason in particular, I figured that we'd want to grab any patch updates for 7.1 so switched to `~>`. Would `< 7.2` do the exact same thing? I can make that switch no problem.

Comment (nefuhs) [reply activerecord-delay_touching.gemspec]: Gotcha, yeah these do the same thing, wasn't too familiar with that syntax.

`"activerecord", "~> 7.1"` works as well.

**Review by nefuhs — APPROVED** [review]

**Review by oehlschl — APPROVED** [review]

**Comment by oehlschl** [comment]
Confirmed that this looks similar to that last time we did this: https://github.com/goatapp/activerecord-delay_touching/pull/7

@jamondrag you should be able to merge this and reference main in your Rails 7.1 branch now.
